### PR TITLE
Updates logger from using Log4j 1.x to 2.x

### PR DESCRIPTION
**Pull request by conniey**
* Updates the test logger from using log4j 1.x to log4j 2.x
  * Log4j 1.x went out of support in 2015 and the last release was in 2012.
* Updates and consolidates logger configuration/use in benchmark and examples project. 
* For benchmarks, there are some configuration values that are passed in at runtime. I assumed they are system arguments that are passed: `-Dazure.cosmos.logger.directory=C:\logs -Dazure.cosmos.hostname=test.cosmosdb.com`.  If they are environment variables, I can update the lookup to use `env` instead of `sys`.

### CODE REVIEW

File: sdk/cosmos/microsoft-azure-cosmos-benchmark/src/test/java/com/azure/data/cosmos/benchmark/LoggerConfigurationTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+public class LoggerConfigurationTest {
+    /**
+     * Netty logging is off. We expect error logging to be disabled.
+     */
+    @Test(groups = {"unit"})
+    public void nettyIsDisabled() {
+        final Logger logger = LoggerFactory.getLogger("io.netty");
+        final Logger logger2 = LoggerFactory.getLogger("io.netty.channel");
+
+        Assert.assertFalse(logger.isErrorEnabled());
```
Comment (moderakh) [inline sdk/cosmos/microsoft-azure-cosmos-benchmark/src/test/java/com/azure/data/cosmos/benchmark/LoggerConfigurationTest.java]: We are using assertj for assertion, please use this instead:
```
import static org.assertj.core.api.Assertions.assertThat;
```

```
assertThat(logger.isErrorEnabled()).isTrue()
```

File: sdk/cosmos/microsoft-azure-cosmos-benchmark/src/test/java/com/azure/data/cosmos/benchmark/LoggerConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    @Test(groups = {"unit"})
+    public void defaultLoggersAtInfo() {
+        final Logger logger = LoggerFactory.getLogger("com.azure.cosmos.internal.query.aggregation");
+        final Logger logger2 = LoggerFactory.getLogger("com.azure.cosmos");
+
+        Assert.assertTrue(logger.isInfoEnabled());
+        Assert.assertTrue(logger2.isInfoEnabled());
+
+        Assert.assertFalse(logger.isDebugEnabled());
+        Assert.assertFalse(logger2.isDebugEnabled());
```
Comment (moderakh) [inline sdk/cosmos/microsoft-azure-cosmos-benchmark/src/test/java/com/azure/data/cosmos/benchmark/LoggerConfigurationTest.java]: ditto

We are using assertj for assertion, please use this instead:
```
import static org.assertj.core.api.Assertions.assertThat;
```

```
assertThat(logger.isErrorEnabled()).isFase()
```

File: sdk/cosmos/microsoft-azure-cosmos/src/test/java/com/azure/data/cosmos/rx/LogLevelTest.java [hunk 3]
```diff
@@ -50,18 +60,30 @@ public void beforeClass() {
...
-        LogManager.getLogger(NETWORK_LOGGING_CATEGORY).setLevel(Level.DEBUG);
-        StringWriter consoleWriter = new StringWriter();
-        WriterAppender appender = new WriterAppender(new PatternLayout(), consoleWriter);
-        LogManager.getLogger(NETWORK_LOGGING_CATEGORY).addAppender(appender);
+        final StringWriter consoleWriter = new StringWriter();
+
+        addAppenderAndLogger(NETWORK_LOGGING_CATEGORY, Level.DEBUG, APPENDER_NAME, consoleWriter);
+
+        Logger logger = LoggerFactory.getLogger(NETWORK_LOGGING_CATEGORY);
+        Assert.assertTrue(logger.isDebugEnabled());
```
Comment (moderakh) [inline sdk/cosmos/microsoft-azure-cosmos/src/test/java/com/azure/data/cosmos/rx/LogLevelTest.java]: here and everywhere else please,

assertj

**Comment by conniey** [comment]
@moderakh Thanks for pointing that out. I've switched to assertj

File: eng/versioning/external_dependencies.txt [hunk 4]
```diff
@@ -66,7 +66,6 @@ io.opentelemetry:opentelemetry-api;0.2.0
 io.opentelemetry:opentelemetry-sdk;0.2.0
 io.projectreactor:reactor-test;3.3.0.RELEASE
 junit:junit;4.13-beta-3
-log4j:log4j;1.2.17
```
Comment (JimSuplizio) [inline eng/versioning/external_dependencies.txt]: Removing this makes me very, very happy...log4j:log4j:1.2.17 is 8 years old.

**Pull request merged**